tests/tensorflow2/test_get_hook.py

```diff
@@ -14,35 +13,43 @@
 def test_get_smdebug_hook_use_smdebug(
     use_smdebug, microbenchmark_repeat, microbenchmark_range, monkeypatch
 ):
-    monkeypatch.setenv("USE_SMDEBUG", use_smdebug)
-    times_taken = []
-    for _ in range(microbenchmark_repeat):
-        timer_context = Timer()
-        with timer_context as t:
-            for _ in range(microbenchmark_range):
-                hook = get_smdebug_hook("keras")
-        times_taken.append(t.time_taken)
+    try:
+        from tensorflow.python.util.smdebug import get_smdebug_hook
+        monkeypatch.setenv("USE_SMDEBUG", use_smdebug)
+        times_taken = []
+        for _ in range(microbenchmark_repeat):
+            timer_context = Timer()
+            with timer_context as t:
+                for _ in range(microbenchmark_range):
+                    hook = get_smdebug_hook("keras")
+            times_taken.append(t.time_taken)
 
-    p95 = percentile(times_taken, 95)
-    # mean time taken with use_smdebug == 0 is 3 seconds
-    # mean time taken with use_smdebug == 1 is 21 seconds
-    threshold = 3 if use_smdebug == "0" else 21
-    assert p95 < threshold
+        p95 = percentile(times_taken, 95)
+        # mean time taken with use_smdebug == 0 is 3 seconds
+        # mean time taken with use_smdebug == 1 is 21 seconds
+        threshold = 3 if use_smdebug == "0" else 21
+        assert p95 < threshold
+    except ImportError:
```

Comment:
What happens if the assert fails on an image that has framework hooks? Right now for all failures it will print "test needs framework hooks" right?

Reply:
Weird. Looked wonderful (and still does) in vi. 
Well cleanse/uniformize indentations.